Fix #9579 - ignore disabled input in Abide

colinmackinlay opened the pull request:
@ncoden To resolve #9579 so that disabled inputs are not validated

colinmackinlay committed:
To resolve #9579

so that disabled inputs are not validated

ncoden commented:
Next time, please use a better name for commit, like `Fix #9579 - ignore disabled input in Abide`

colinmackinlay commented:
Sorry about that - realised after I'd moved longer description to the body.

Tried to update docs but can't see how to edit "Javascript Reference" which seems the most appropriate place. The abide.md file doesn't include this or the SASS reference. Presumably this is generated elsewhere?

ncoden commented:
"Javascript Reference" is generated from the comments above functions and options, like [here](https://github.com/ncoden/foundation-sites/blob/a87eb5dc02f714f80bd3648151d2c53185827548/js/foundation.abide.js#L84). But I think we should also add a section to the docs (like `Ignored inputs` with an example for `[data-abide-ignore]`, `[type="hidden"]` and [disabled]`).

colinmackinlay committed:
'Ignore' details added to  validateInput() comment

colinmackinlay committed:
Update abide.md with extra ignore conditions

In addition to data-abide-ignore, abide also ignores type="hidden" attribute and in this patch the diabled attribute

ncoden commented on docs/pages/abide.md:
```diff
@@ -34,6 +34,18 @@ These input types create a text field: `text`, `date`, `datetime`, `datetime-loc
     </div>
     <div class="small-12 columns">
+      <label>Disabled!
+        <input type="text" placeholder="You can't use me!" aria-describedby="exampleHelpTex" disabled>
```
`placeholder="Disabled input"`

ncoden replied:
Also, don't you think there is already too much inputs in this example, and we should rather move ignored input to their own section ?

ncoden commented on docs/pages/abide.md:
```diff
@@ -34,6 +34,18 @@ These input types create a text field: `text`, `date`, `datetime`, `datetime-loc
+    </div>
+    <div class="small-12 columns">
+      <label>Hidden!
+        <input type="hidden" placeholder="You can't see me!" aria-describedby="exampleHelpTex" >
```
placeholder="Hidden input"

colinmackinlay commented:
validateInput() comment updated and I've added two further examples below the data-abide-ignore one.
Perhaps the [type="hidden"] one could be clearer but it's hard to see how!

colinmackinlay committed:
Moved ignore examples to separate section

colinmackinlay commented:
Section before Initial State seems to just hang without explanation. It seems to be an example of how to show error state initially both in the label and with callout. Is that right @gakimball ?

kball commented:
Merged!  Thanks @colinmackinlay
